src/base_cmdline.cpp

```diff
@@ -1,235 +1,231 @@
+		{"o", "output", "Control what parts of the output to display (default:all)", false, false},
+		{"q", "quiet", "Silence progress reporting", true, false},
+		{"un", "unicode", "Allow unicode output", true, false},
+		{"v", "version", "Display the app version", true, false}
+	};
+
+	std::map<std::string, std::string> parsedOptions;
+	std::map<std::string, bool> parsedFlags;
+
+public:
+	bool parse(int argc, char* argv[]) {
+	    for (int i = 1; i < argc; i++) {
+	        std::string arg = argv[i];
+	        if (arg[0] != '-') continue;
+
+	        std::string optName = arg.substr(arg[1] == '-' ? 2 : 1);
+	        auto option = findOption(optName);
+	        if (!option) {
+	            std::cerr << "Unknown option: " << arg << std::endl;
+	            return false;
+	        }
+
+	        if (option->isSwitch) {
+	            parsedFlags[option->longName] = true;
+	        } else if (i + 1 < argc) {
+	            parsedOptions[option->longName] = argv[++i];
+	        } else {
+	            std::cerr << "Missing value for option: " << arg << std::endl;
+	            return false;
+	        }
+	    }
+
+	    // Don't validate required options for help or version flags
+	    if (parsedFlags.count("help") > 0 || parsedFlags.count("version") > 0) {
+	        return true;
+	    }
+
+	    return validateRequiredOptions();
+	}
+
+	bool found(const std::string& name) const {
+		return parsedFlags.count(name) > 0 || parsedOptions.count(name) > 0;
+	}
+
+	std::optional<std::string> getValue(const std::string& name) const {
+		auto it = parsedOptions.find(name);
+		return it != parsedOptions.end() ? std::optional<std::string>(it->second) : std::nullopt;
+	}
+
+	void displayHelp() const {
+		std::cout << "Usage:\n";
+		for (const auto& opt : options) {
+			std::cout << "  -" << opt.shortName << ", --" << opt.longName
+					 << (opt.isRequired ? " (required)" : "") << "\n    "
+					 << opt.description << "\n";
+		}
+	}
+
+private:
+	std::optional<CommandLineOption> findOption(const std::string& name) const {
+		for (const auto& opt : options) {
+			if (opt.shortName == name || opt.longName == name) {
+				return opt;
+			}
+		}
+		return std::nullopt;
+	}
+
+	bool validateRequiredOptions() const {
+		std::vector<std::string> missing;
+		for (const auto& opt : options) {
+			if (opt.isRequired && !found(opt.longName)) {
+				missing.push_back(opt.longName);
+			}
+		}
+
+		if (!missing.empty()) {
+			Ctrl::getInstance()->displayErrorMessage(910 + missing.size(), missing);
+			return false;
+		}
+		return true;
+	}
+};
 
-bool validateProbes(const double r1, const double r2, const bool pm){
-  if(pm && r2 < r1){
-    Ctrl::getInstance()->displayErrorMessage(104);
-    return false;
-  }
-  return true;
+// Validation functions combined into single-line expressions
+bool validateProbes(double r1, double r2, bool pm) {
+	return !(pm && r2 < r1 && (Ctrl::getInstance()->displayErrorMessage(104), true));

```

Comment:
Please avoid changing things that are already working as intended (even if it improves the code). It risks introducing bugs. Are these functions covered by unit tests? Otherwise I'll have to test them myself.